cmd/auth-provider-gcp/app/getcredentials.go

```diff
@@ -92,13 +128,13 @@ func getCredentials(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func defineFlags(credCmd *cobra.Command) {
-	credCmd.Flags().StringVarP(&authFlow, "authFlow", "a", gcrAuthFlow, "authentication flow (valid values are gcr, dockercfg, and dockercfg-url)")
+func defineFlags(credCmd *cobra.Command, options *CredentialOptions) {
+	credCmd.Flags().StringVarP(&options.AuthFlow, "authFlow", "a", gcrAuthFlow, "authentication flow (valid values are gcr, dockercfg, and dockercfg-url)")
```

Comment:
Earlier we have code which says

"(must be one of %q, %q, or %q)", gcrAuthFlow, dockerConfigAuthFlow, dockerConfigURLAuthFlow

Having this and that seems inconsistent. The referenced code is nice in that is ensures that the strings checked are the same as the string shown to the customer. It also reduces the number of places we have to change if we need to fix a constant.

Reply:
Good catch, done